src/python/pants/testutil/rule_runner.py

```diff
@@ -75,6 +75,53 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class EngineErrorTrap:
+    """A context manager to catch `ExecutionError`s in tests and get the underlying exception.
+
+    Use like this:
+
+        with EngineErrorTrap() as trap:
+            rule_runner.request(OutputType, [input])
+        assert isinstance(trap.e, ValueError)
```

Comment:
Perhaps add two fields, like `pytest.raises` to support optionally passing in expected exception and message in the constructor?

```python
with EngineErrorTrap(expected_exception=ProcessExecutionFailure, message_contains="SECURITY ERROR"):
    rule_runner.request(...)
```

Which are asserted during `__exit__`.

I like that aspect of `pytest.raises`, but in case that is an undesirable style, then please ignore ;)

Reply:
I love that recommendation, that will cut down on boilerplate even more!